File: Java/AlarmsProcedural/src/main/java/Main/Building.java
package Main;

import AlarmServices.AlarmService;

public class Building {
  private final Alarm _alarm;

  public Building(Alarm alarm) {
    _alarm = alarm;
  }

  public void someoneEntered() {
    AlarmService.triggerAlarm(_alarm);
  }

  public void allClear() {
    AlarmService.disableAlarm(_alarm);
  }

  public void dump() {
    AlarmService.dump(_alarm);
  }
}

File: CSharp/TellDontAsk/AlarmsProcedural/Building.cs
using AlarmsProcedural.AlarmServices;

namespace AlarmsProcedural
{
  public class Building
  {
    private readonly Alarm _alarm;

    public Building(Alarm alarm)
    {
      _alarm = alarm;
    }

    public void SomeoneEntered()
    {
      AlarmService.TriggerAlarm(_alarm);
    }

    public void AllClear()
    {
      AlarmService.DisableAlarm(_alarm);
    }

    public void Dump()
    {
      AlarmService.Dump(_alarm);
    }
  }
}